Source/Core/Core/Core.cpp

```diff
@@ -346,11 +347,22 @@ static void CpuThread(const std::optional<std::string>& savestate_path, bool del
   }
 #endif
 
+  GDBThread gdb_thread;
+  if (_CoreParameter.bEnableDebugging)
+  {
+    gdb_thread.Initialize();
+  }
+
   // Enter CPU run loop. When we leave it - we are done.
   CPU::Run();
 
   s_is_started = false;
 
+  if (_CoreParameter.bEnableDebugging)
```

Comment:
Thoughts on checking whether `gdb_thread` is valid/alive instead of checking the parameter? Not quite sure what happens if (or when) that value changes after the CPU is started.

Reply:
True.  Some of the logic will need to be updated to take that possibility into account.  Since i had always been using the command line parameter which turns this on, did not remember about the fact it was exposed from within the UI as well. Will take a look and update accordingly.